### PR TITLE
Add documentation to embed an nREPL, specifically for Rebel Readline

### PR DESCRIPTION
**Pull request by Futurile**
Rebel Readline (https://github.com/bhauman/rebel-readline) is a user-friendly REPL that goes well with vim-iced. It only has a local service, so you have to create an nREPL after it's been started. This means you have to embed an nREPL. 

 It would be nicer to create an nREPL service in Rebel Readline, but it's too complicated!

Added documentation and example 'user.clj'.

@agriffis came up with this solution at https://github.com/clojure-emacs/cider-nrepl/issues/447#issuecomment-570826543

### CODE REVIEW

File: doc/vim-iced.txt [hunk 1]
```diff
@@ -217,6 +218,48 @@ SHADOW-CLJS~
 <
   To start CLJS REPL, see |vim-iced-shadow-cljs|.
 
+                                              *vim-iced-manual-rebel-readline*
+REBEL READLINE~
+
+  If you use Rebel Readline or want to embed an nREPL in your application,
+  then one approach is to create a `user.clj` with the following:
+>
```
Comment (liquidz) [inline doc/vim-iced.txt:226]:
```suggestion
>
```

File: doc/vim-iced.txt [hunk 2]
```diff
@@ -217,6 +218,48 @@ SHADOW-CLJS~
...
   To start CLJS REPL, see |vim-iced-shadow-cljs|.
 
+                                              *vim-iced-manual-rebel-readline*
+REBEL READLINE~
+
+  If you use Rebel Readline or want to embed an nREPL in your application,
+  then one approach is to create a `user.clj` with the following:
+>  
+  (ns user (:require [nrepl.server :as nrepl-server]
+                  [clojure.java.io :as io]))
```
Comment (liquidz) [inline doc/vim-iced.txt:228]:
```suggestion
  (ns user
    (:require [nrepl.server :as nrepl-server]
              [clojure.java.io :as io]))
```

File: doc/vim-iced.txt [hunk 3]
```diff
@@ -217,6 +218,48 @@ SHADOW-CLJS~
...
+
+  (defn dev-middleware []
+    (mapcat (fn [[ns syms]] (require ns) (map (partial ns-resolve ns) syms))
+    [['refactor-nrepl.middleware ['wrap-refactor]] ['iced.nrepl ['wrap-iced]]]))
+   
+  (defn start-nrepl-server! []
+    (reset!
+      nrepl-server
+        (nrepl-server/start-server :port nrepl-port :handler (nrepl-handler))
+    )
```
Comment (liquidz) [inline doc/vim-iced.txt:248]:
```suggestion
        (nrepl-server/start-server :port nrepl-port :handler (nrepl-handler)))
```

File: doc/vim-iced.txt [hunk 4]
```diff
@@ -217,6 +218,48 @@ SHADOW-CLJS~
...
+    (spit ".nrepl-port" nrepl-port))
+
+  (defn stop-nrepl-server! []
+    (when (not (nil? @nrepl-server))
+      (nrepl-server/stop-server @nrepl-server)
+      (println "Cider nREPL server on port" nrepl-port "stopped")
+      (reset! nrepl-server nil)
+      (io/delete-file ".nrepl-port" true)
+      (System/exit 0)))
+<
```
Comment (liquidz) [inline doc/vim-iced.txt:259]:
```suggestion
<
```

File: doc/vim-iced.txt [hunk 5]
```diff
@@ -217,6 +218,48 @@ SHADOW-CLJS~
...
+  
+  (def nrepl-port 7888)
+  (defonce nrepl-server (atom nil))
+
+  (defn cider-middleware
+    "Get cider middleware, see 
+    https://github.com/clojure-emacs/cider-nrepl/issues/447"
+    []
+    (require 'cider.nrepl)
+    (map resolve (var-get (ns-resolve 'cider.nrepl 'cider-middleware))))
```
Comment (liquidz) [inline doc/vim-iced.txt:238]: `cider-middleware` contains all middlewares provided from cider-nrepl.
https://github.com/clojure-emacs/cider-nrepl/blob/v0.25.4/src/cider/nrepl/middleware.clj#L7

Now `track-state` middleware may lead vim-iced's hang-up. (c.f. #170)
So, to be safe, we should specify the middleware separately, as in Leiningen's example.

File: doc/vim-iced.txt [hunk 6]
```diff
@@ -217,6 +218,48 @@ SHADOW-CLJS~
...
+
+  (defn dev-middleware []
+    (mapcat (fn [[ns syms]] (require ns) (map (partial ns-resolve ns) syms))
+    [['refactor-nrepl.middleware ['wrap-refactor]] ['iced.nrepl ['wrap-iced]]]))
+   
+  (defn start-nrepl-server! []
+    (reset!
+      nrepl-server
+        (nrepl-server/start-server :port nrepl-port :handler (nrepl-handler))
```
Comment (liquidz) [inline doc/vim-iced.txt:247]: Are you missing the definition of `nrepl-handler`?

**Review by liquidz — CHANGES_REQUESTED** [review]
@Futurile Thanks a lot!
Cound you confirm my review comments?

File: doc/vim-iced.txt (hunk 5 above)
Comment (bbatsov) [reply doc/vim-iced.txt:239]: Btw, now that https://github.com/clojure-emacs/cider-nrepl/issues/447 is fixed, this `resolve` step is not necessary.

Comment (liquidz) [reply doc/vim-iced.txt:239]: Thank you for the information!

File: doc/vim-iced.txt (hunk 6 above)
Comment (Futurile) [reply doc/vim-iced.txt]: @liquidz you were right, I missed it out somehow. Updated the PR.

File: doc/vim-iced.txt (hunk 5 above)
Comment (Futurile) [reply doc/vim-iced.txt]: @liquidz My poor understanding of Clojure and cider are showing here. I believe you wanted it to specifically load the cider middleware that you have for Leiningen. I have done this - probably in the ugliest way possible. 

I removed the resolve step as bbatsov suggested.

**Review by liquidz — APPROVED** [review]
@Futurile Thanks!
LGTM

**Pull request merged**